triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -456,4 +456,48 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
       } yield succeed
   }
 
+  it should "stop a trigger when the user script fails init" in withHttpService(Some(dar)) {
+    (uri: Uri, client: LedgerClient, ledgerProxy: Proxy) =>
+      for {
+        resp <- startTrigger(uri, s"$testPkgId:ErrorTrigger:trigger", "Alice")
+        aliceTrigger <- parseTriggerId(resp)
+        _ <- assertTriggerStatus(
+          uri,
+          aliceTrigger,
+          _ ==
+            Vector(
+              "starting",
+              "stopped: initialization failure",
```

Comment:
Why does this result in an initialization failure instead of a runtime failure? Does the trigger `initialize` step fail if the `triggerRule` throws an `error`?

Reply:
It's possible for `getTriggerSink` to throw an exception in the evaluation of `runWithACS`. When that happens, it does not manifest as a failing future which is where we expect `RuntimeException`s but is immediate hence the `try`/`catch` and categorization as an `InitializationFailure`.